### PR TITLE
common: remove usage of deprecated function

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
This changes the default behaviour of ToHexArray and with it the behaviour of eth_getProof.
Previously we encoded an empty slice as 0, now the empty slice is encoded as 0x.

~~Also it introduces the hexutil package into common so it might not be so good after all~~

### CODE REVIEW

**Comment by holiman** [comment]
> Also it introduces the hexutil package into common so it might not be so good after all

yeah, if there's no such dep right now, better avoid it (copy-paste on a small function is preferrable over adding a module dep that isn't already there, IMO)

**Comment by mrFranklin** [comment]
@MariusVanDerWijden @holiman 
That is the reason why I think the method is not `Deprecated`
So Could we merge my PR?  #21609

**Comment by karalabe** [comment]
I'd vote for revisiting this. If the problem is that we don' want to introuduce `hexutil` into `common`, then IMHO we should move `ToHexArray` (or a variation of the name) into `hexutil`. That would be a better place either way.

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -592,14 +592,23 @@ func (s *PublicBlockChainAPI) GetProof(ctx context.Context, address common.Addre
 		codeHash = crypto.Keccak256Hash(nil)
 	}
 
+	// toHexArray creates a array of hex-string based on []byte
```
Comment (fjl) [inline internal/ethapi/api.go:595]: I would prefer this to be a globally-defined unexported function in this package.

**Comment by MariusVanDerWijden** [comment]
cc @fjl

**Review by fjl — APPROVED** [review]

**Pull request merged**